addons/im_livechat/models/discuss_channel_member_history.py

```diff
@@ -0,0 +1,34 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import models, fields
+
+
+class DiscussChannelMemberHistory(models.Model):
+    _name = "discuss.channel.member.history"
+    _description = "Keep the channel member history"
+
+    channel_id = fields.Many2one("discuss.channel")
+    partner_id = fields.Many2one("res.partner")
+    guest_id = fields.Many2one("mail.guest")
+    member_id = fields.Many2one("discuss.channel.member", ondelete="set null")
+    type = fields.Selection([
+        ("operator", "Operator"),
+        ("visitor", "Visitor"),
+        ("bot", "Chat bot"),
+    ])
+
+    def create_from_members(self, members):
+        create = []
+        for member in members:
+            if not member.channel_member_history_ids:
+                type = "visitor"
+                if member.partner_id:
+                    type = "operator" if not member.partner_id.chatbot_script_id else "bot"
+                create += [{
```

Comment:
might as well just use `append` instead of creating an array just to concat it :)